pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -328,6 +366,17 @@ func injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo) error {
 
 	injectLibVolume(pod)
 
+	// If, by this point, DD_SERVICE has not been set and apm_instrumentation_enabled is true then set any config
```

Comment:
I am not sure to understand the logic there. There's no check about whether `DD_SERVICE` is set or not, isn't it? It's just that `injectEnv` does not override existing variables or am I missing something?

Also if I understood correctly, `injectLibConfig` never injects anything except if `.config.v1` annotation is set, in which case, something has already been injected, but if `apm_instrumentation_enabled` is set, we go through, try to inject things, just for `injectEnv` to skip setting because already existing?

Also, as currently in most cases, the config is not injected, what's the current default config (e.g. no variables set), is there a difference with `basicConfig`?

Reply:
> I am not sure to understand the logic there. There's no check about whether DD_SERVICE is set or not, isn't it? It's just that injectEnv does not override existing variables or am I missing something?

Ah yes, this comment is stale now. We were originally only injecting DD_SERVICE. Correct that injectEnv does not override existing variables. I will update this comment.

> Also if I understood correctly, injectLibConfig never injects anything except if .config.v1 annotation is set, in which case, something has already been injected, but if apm_instrumentation_enabled is set, we go through, try to inject things, just for injectEnv to skip setting because already existing?

Yes, we are simply trying to inject env vars that haven't already been added by the user (provide defaults). Previously the only way configuration could be added here is via the .config.v1 annotation (which users are not expected to add themselves).

> Also, as currently in most cases, the config is not injected, what's the current default config (e.g. no variables set), is there a difference with basicConfig?

If I follow you correctly, previously there was no default config. When remote instrumentation is applied (.config.v1 annotation set on the pod) it specifies the user configuration done through the Datadog UI/RC. For users [manually adding the annotations](https://docs.datadoghq.com/tracing/trace_collection/library_injection_local/?tab=kubernetes) they are expected to set the tracer env vars themselves.

This means the "default config" was that of the settings defined by each tracer which are not consistent. To get consistency between all the libraries @jonbodner has come up with what the basicConfig defaults.